pkg/agent/provisioning.go

```diff
@@ -34,46 +28,118 @@ func createOrUpdateCredentialRequestDefinition(data *v1alpha1.CredentialRequestD
 	return data, nil
 }
 
+type crdBuilderOptions struct {
+	name      string
+	provProps []provisioning.PropertyBuilder
+	reqProps  []provisioning.PropertyBuilder
+}
+
 // NewCredentialRequestBuilder - called by the agents to build and register a new credential reqest definition
-func NewCredentialRequestBuilder() provisioning.CredentialRequestBuilder {
-	return provisioning.NewCRDBuilder(createOrUpdateCredentialRequestDefinition)
+func NewCredentialRequestBuilder(options ...func(*crdBuilderOptions)) provisioning.CredentialRequestBuilder {
```

Comment:
A specific type might be helpful for these funcs

Reply:
Actually, I'm not sure how an agent can implement this func because the param is a private type

Reply:
They use the With functions that return that function. They do not need to know that type.

Reply:
I see a few exported funcs like `WithCRDProvisionSchemaProperty ` that return `func(c *crdBuilderOptions)` that an agent can use. But if an agent needs their own option func then they won't be able to implement it. Not sure if they need it or not, just bringing it up.

Reply:
Nah, I think we will be the only ones providing these options